POH to PreservedCopy refactoring

jmartin-sul opened the pull request:
* move `POH#update_pc_audit_timestamps` into the `PreservedCopy` model, fix existing tests
* move `POH#update_preserved_copy_version_etc` into `PreservedCopy` model and rename to ~~`update_version_and_size`~~ `upd_audstamps_version_size`. fix existing tests.
* move part of `POH#update_status` into the `PreservedCopy` model, fix existing tests
* new tests for methods that were refactored from `POH` to `PreservedCopy`

paired w/ @SaravShah 

closes #470 
connects to #476 (now that we have methods to more easily update PC audit timestamps, we should be able to get the C2M code to update those timestamps, and thus implement the test mentioned in #476)
closes #370

jmartin-sul commented on app/models/preserved_copy.rb:
```diff
@@ -30,4 +30,24 @@ class PreservedCopy < ApplicationRecord
+    update_audit_timestamps(ran_moab_validation, true)
+  end
+
+  # if status has actually changed, the provided block will be executed
```
yeah, i was on the fence about that one, happy to remove

jmartin-sul commented on spec/models/preserved_copy_spec.rb:
```diff
@@ -86,4 +86,98 @@
+    it 'updates version' do
+      pc.upd_audstamps_version_size(false, 3, nil)
+      expect(pc.version).to eq 3
+    end
```
we were just keeping the logic as-is from what we extracted.  in POH, version is a required param, and on the model it's a required field, so i think we don't have to account for version coming in as nil here... unless we thought this newly moved method might be called from other places with a nil version, and we wanted to make sure we provided a more helpful error message (than AR barfing when nil gets set on a required field).

ndushay replied:
sure.  Should the method comments indicate it expects version not to be nil?  Or just leave it?

jmartin-sul commented:
made all of the changes suggested by @ndushay, should be ready for re-review.

i did a bit of commit history cleanup, and i rebased on what's now current master, but here's the diff of my local branch vs what was on GH before i rebased on master and re-pushed:
```sh
$ git diff origin/poh-to-pc-refactoring 
diff --git a/app/models/preserved_copy.rb b/app/models/preserved_copy.rb
index 362ab34..0c7450d 100644
--- a/app/models/preserved_copy.rb
+++ b/app/models/preserved_copy.rb
@@ -31,20 +31,21 @@ class PreservedCopy < ApplicationRecord
   validates :status, inclusion: { in: statuses.keys }
   validates :version, presence: true
 
-  def update_audit_timestamps(ran_moab_validation, version_audited)
+  def update_audit_timestamps(moab_validated, version_audited)
     t = Time.current
-    self.last_moab_validation = t if ran_moab_validation
+    self.last_moab_validation = t if moab_validated
     self.last_version_audit = t if version_audited
   end
 
-  # expects @incoming_version to be numeric
-  def upd_audstamps_version_size(ran_moab_validation, new_version, new_size)
+  # moab_validated is a boolean indicating whether validation has been run (regardless of the result)
+  # new_version is expected to be numeric
+  # new_size is expected to be numeric if provided (nil is allowed)
+  def upd_audstamps_version_size(moab_validated, new_version, new_size)
     self.version = new_version
     self.size = new_size if new_size
-    update_audit_timestamps(ran_moab_validation, true)
+    update_audit_timestamps(moab_validated, true)
   end
 
-  # if status has actually changed, the provided block will be executed
   def update_status(new_status)
     return unless new_status != status
     yield
diff --git a/spec/models/preserved_copy_spec.rb b/spec/models/preserved_copy_spec.rb
index d8228ce..e8c2e48 100644
--- a/spec/models/preserved_copy_spec.rb
+++ b/spec/models/preserved_copy_spec.rb
@@ -98,12 +98,12 @@ RSpec.describe PreservedCopy, type: :model do
       )
     end
 
-    it 'updates last_moab_validation time if ran_moab_validation is true' do
+    it 'updates last_moab_validation time if moab_validated is true' do
       expect(pc.last_moab_validation).to be nil
       pc.update_audit_timestamps(true, false)
       expect(pc.last_moab_validation).not_to be nil
     end
-    it 'does not update last_moab_validation time if ran_moab_validation is false' do
+    it 'does not update last_moab_validation time if moab_validated is false' do
       expect(pc.last_moab_validation).to be nil
       pc.update_audit_timestamps(false, false)
       expect(pc.last_moab_validation).to be nil
```

which should just be the set of changes requested in review.

jmartin-sul commented:
@tallenaz: yeah, `upd_audstamps_version_size` was named that way for brevity.  i think the non-abbreviated version would be something like `update_auditstamps_version_and_size`.  which i'd be fine with.

any input from @ndushay, who i think prizes brevity in names more than i do?

ndushay commented:
oddly, i don't have strong feelings about that method name.  If we wanted shorter, it could be maybe:

`update_version_check_attributes` or `update_version_audit_attributes` or something equally weird.

I'm okay with any in this case.

ndushay commented on app/models/preserved_copy.rb:
```diff
@@ -30,4 +30,25 @@ class PreservedCopy < ApplicationRecord
+
+  # moab_validated must not be nil. boolean indicating whether validation has been run (regardless of result).
+  # new_version is expected to be numeric
+  # new_size is expected to be numeric if provided (nil is allowed)
```
super picky / not required:  I'm used to the @param notation in comments.

jmartin-sul replied:
ah yeah, good point, filed #490 for a bit of follow on work in this model
